File: phone-exchange-2-dot-net-examples/GetExchangeInfoRestSdkExample.cs
using phone_exchange_2_dot_net.REST;

namespace phone_exchange_2_dot_net_examples
{
    public class GetExchangeInfoRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n-------------------------------------------");
            Console.WriteLine("PhoneExchange2 - GetExchangeInfo - REST SDK");
            Console.WriteLine("-------------------------------------------");

            GetExchangeInfoClient.GetExchangeInfoInput getExchangeInfoInput = new(
                PhoneNumber: "805-555-1212",
                CountryCode: "1",
                Country: "US",
                IPAddress: "",
                CallerCountry: "",
                Extras: "",
                Token: "",
                LicenseKey: licenseKey,
                IsLive: isLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Phone Number  : {getExchangeInfoInput.PhoneNumber}");
            Console.WriteLine($"Country Code  : {getExchangeInfoInput.CountryCode}");
            Console.WriteLine($"Country       : {getExchangeInfoInput.Country}");
            Console.WriteLine($"IP Address    : {getExchangeInfoInput.IPAddress}");
            Console.WriteLine($"Caller Country: {getExchangeInfoInput.CallerCountry}");
            Console.WriteLine($"Extras        : {getExchangeInfoInput.Extras}");
            Console.WriteLine($"Token         : {getExchangeInfoInput.Token}");
            Console.WriteLine($"License Key   : {getExchangeInfoInput.LicenseKey}");
            Console.WriteLine($"Is Live       : {getExchangeInfoInput.IsLive}");

            PE2Response response = GetExchangeInfoClient.Invoke(getExchangeInfoInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Exchange Info *\r\n");
                if (response.ExchangeInfoResults?.Length > 0)
                {
                    foreach (var exchangeInfo in response.ExchangeInfoResults)
                    {
                        Console.WriteLine($"Phone Number     : {exchangeInfo.PhoneNumber}");
                        Console.WriteLine($"Name             : {exchangeInfo.Name}");
                        Console.WriteLine($"City             : {exchangeInfo.City}");
                        Console.WriteLine($"State            : {exchangeInfo.State}");
                        Console.WriteLine($"Country          : {exchangeInfo.Country}");
                        Console.WriteLine($"Line Type        : {exchangeInfo.LineType}");
                        Console.WriteLine($"Time Zone        : {exchangeInfo.TimeZone}");
                        Console.WriteLine($"Latitude         : {exchangeInfo.Latitude}");
                        Console.WriteLine($"Longitude        : {exchangeInfo.Longitude}");
                        Console.WriteLine($"SMS Address      : {exchangeInfo.SMSAddress}");
                        Console.WriteLine($"MMS Address      : {exchangeInfo.MMSAddress}");
                        Console.WriteLine($"Note Codes       : {exchangeInfo.NoteCodes}");
                        Console.WriteLine($"Note Descriptions: {exchangeInfo.NoteDescriptions}");

                        Console.WriteLine("\r\n* Ported Info *\r\n");
                        if (exchangeInfo.PortedInfo != null)
                        {
                            Console.WriteLine($"Original Name     : {exchangeInfo.PortedInfo.OriginalName}");
                            Console.WriteLine($"Original Line Type: {exchangeInfo.PortedInfo.OriginalLineType}");
                            Console.WriteLine($"Ported Date       : {exchangeInfo.PortedInfo.PortedDate}");
                            Console.WriteLine($"LATA              : {exchangeInfo.PortedInfo.LATA}");
                        }
                        else
                        {
                            Console.WriteLine("No ported info found.");
                        }
                    }
                }
                else
                {
                    Console.WriteLine("No exchange info items found.");
                }
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: phone-exchange-2-nodejs-examples/get_exchange_info_rest_sdk_example.js
import { GetExchangeInfoClient } from '../phone-exchange-2-nodejs/REST/get_exchange_info_rest.js';

async function GetExchangeInfoRestGo(licenseKey, isLive) {
    console.log('\n-------------------------------------------');
    console.log('PhoneExchange2 - GetExchangeInfo - REST SDK');
    console.log('-------------------------------------------');

    const phoneNumber = '+18059631700';
    const timeoutSeconds = 15;

    console.log('\n* Input *\n');
    console.log(`Phone Number   : ${phoneNumber}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetExchangeInfoClient.invoke(phoneNumber, licenseKey, isLive, timeoutSeconds);

        if (response.Error) {
            console.log('\n* Error *\n');
            console.log(`Error Type    : ${response.Error.Type}`);
            console.log(`Error DescCode: ${response.Error.DescCode}`);
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error TypeCode: ${response.Error.TypeCode}`);
        }

        console.log('\n* Exchange Info *\n');
        if (response.ExchangeInfoResults && response.ExchangeInfoResults.length > 0) {
            for (const exchangeInfo of response.ExchangeInfoResults) {
                console.log(`Phone Number     : ${exchangeInfo.PhoneNumber}`);
                console.log(`Carrier Name     : ${exchangeInfo.Name}`);
                console.log(`City             : ${exchangeInfo.City}`);
                console.log(`State            : ${exchangeInfo.State}`);
                console.log(`Country          : ${exchangeInfo.Country}`);
                console.log(`Line Type        : ${exchangeInfo.LineType}`);
                console.log(`Time Zone        : ${exchangeInfo.TimeZone}`);
                console.log(`Latitude         : ${exchangeInfo.Latitude}`);
                console.log(`Longitude        : ${exchangeInfo.Longitude}`);
                console.log(`SMS Address      : ${exchangeInfo.SMSAddress}`);
                console.log(`MMS Address      : ${exchangeInfo.MMSAddress}`);
                console.log(`Note Codes       : ${exchangeInfo.NoteCodes}`);
                console.log(`Note Descriptions: ${exchangeInfo.NoteDescriptions}`);

                console.log('\n* Ported Info *\n');
                if (exchangeInfo.PortedInfo) {
                    console.log(`Original Carrier Name: ${exchangeInfo.PortedInfo.OriginalName}`);
                    console.log(`Original Line Type   : ${exchangeInfo.PortedInfo.OriginalLineType}`);
                    console.log(`Ported Date          : ${exchangeInfo.PortedInfo.PortedDate}`);
                    console.log(`LATA                 : ${exchangeInfo.PortedInfo.LATA}`);
                } else {
                    console.log('No ported info found.');
                }
            }
        } else {
            console.log('No exchange info results found.');
        }
    } catch (error) {
        console.log('\n* Error *\n');
        console.log(`Error Message: ${error.message}`);
    }
}

export { GetExchangeInfoRestGo };